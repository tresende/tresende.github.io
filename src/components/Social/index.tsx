import SocialLink from 'components/SocialLink'

const Social = () => (
  <footer className="flex content-between justify-center gap-2">
    <SocialLink url="https://github.com/tresende" name="github" />
    <SocialLink url="https://www.linkedin.com/in/thiago-resende-46175391" name="linkedin" />
    <SocialLink url="mailto:thiago.gcresende@gmail.com" name="email" />
    <SocialLink url="https://www.instagram.com/thiago.gcr/" name="instagram" />
  </footer>
)

export default Social
