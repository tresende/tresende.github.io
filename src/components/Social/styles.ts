import styled from 'styled-components'

export const Wrapper = styled.footer`
  display: flex;
  justify-content: space-between;
  margin: 0 auto;
  width: 12rem;
`
