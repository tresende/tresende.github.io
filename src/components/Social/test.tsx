import { render, screen } from '@testing-library/react'

import Social from '.'

describe('<Social />', () => {
  it('should render 4 social links', () => {
    const { container } = render(<Social />)

    expect(screen.getAllByRole('link')).toHaveLength(4)
    expect(container).toMatchSnapshot()
  })
})
