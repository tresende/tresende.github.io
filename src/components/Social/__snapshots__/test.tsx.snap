// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Social /> should render 4 social links 1`] = `
<div>
  <footer
    class="flex content-between justify-center gap-2"
  >
    <a
      href="https://github.com/tresende"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="github icon"
        height="32"
        src="/img/github.png"
        width="32"
      />
    </a>
    <a
      href="https://www.linkedin.com/in/thiago-resende-46175391"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="linkedin icon"
        height="32"
        src="/img/linkedin.png"
        width="32"
      />
    </a>
    <a
      href="mailto:thiago.gcresende@gmail.com"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="email icon"
        height="32"
        src="/img/email.png"
        width="32"
      />
    </a>
    <a
      href="https://www.instagram.com/thiago.gcr/"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="instagram icon"
        height="32"
        src="/img/instagram.png"
        width="32"
      />
    </a>
  </footer>
</div>
`;
